pkg/where-resolver/reconcile_on_location.go

```diff
@@ -291,20 +320,56 @@ func cleanSPSByLoc(sps *edgev2alpha1.SinglePlacementSlice, lws, lName string) *e
 	return sps
 }
 
-func makeSinglePlacementsForLoc(locSelectingSts *edgev2alpha1.Location, sts []*edgev2alpha1.SyncTarget) []edgev2alpha1.SinglePlacement {
+func (c *controller) makeSinglePlacementsForLoc(locSelectingSts *edgev2alpha1.Location, sts []*edgev2alpha1.SyncTarget) []edgev2alpha1.SinglePlacement {
 	made := []edgev2alpha1.SinglePlacement{}
 	if locSelectingSts == nil || len(sts) == 0 {
 		return made
 	}
-	ws := logicalcluster.From(locSelectingSts).String()
+	_, locOriginalName, kbSpaceID, err := kbuser.AnalyzeObjectID(locSelectingSts)
+	if err != nil {
+		return made
+	}
+	locSpaceID := c.kbSpaceRelation.SpaceIDFromKubeBind(kbSpaceID)
+	if locSpaceID == "" {
+		return made
+	}
 	for _, st := range sts {
+		_, stOriginalName, _, err := kbuser.AnalyzeObjectID(st)
+		if err != nil {
+			continue
+		}
 		sp := edgev2alpha1.SinglePlacement{
-			Cluster:        ws,
-			LocationName:   locSelectingSts.Name,
-			SyncTargetName: st.Name,
+			Cluster:        locSpaceID,
+			LocationName:   locOriginalName,
+			SyncTargetName: stOriginalName,
 			SyncTargetUID:  st.UID,
 		}
 		made = append(made, sp)
 	}
 	return made
 }
+
+func (c *controller) getConsumerSpaceForSPS(sps *edgev2alpha1.SinglePlacementSlice) (string, string, error) {
```

Comment:
The two errors are not alike, should be handled differently (first one does not benefit from retry, the second one does).

Reply:
This is part of the error fusion noted above and can be should be fixed in follow-on work that fixes the error fusion issue in general.